eo-maven-plugin/src/main/java/org/eolang/maven/RustProject.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2023 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.eolang.maven.footprint.Footprint;
+import org.eolang.maven.footprint.FtDefault;
+
+/**
+ * To create the cargo project.
+ * @since 1.0
+ * @todo #1877:90min Transform source rust file by changing name of
+ *  function in it to jni standard. It also necessary to add required
+ *  headers and imports related to jni.
+ * @todo #1877:90min Build the rust project via 'cargo build' command.
+ *  Thins command must be called from right directory. Think of correct
+ *  compilation errors handling.
+ */
+public class RustProject {
+    /**
+     * Path to cargo project.
+     */
+    private final Path dest;
+
+    /**
+     * Footprint.
+     */
+    private final Footprint footprint;
+
+    /**
+     * Collection of dependencies of the project.
+     */
+    private final Set<String> dependencies;
+
+    /**
+     * Ctor.
+     * Creates a raw cargo project.
+     * @param target Destination path.
+     * @throws IOException If any issues with I/O
+     */
+    @SuppressWarnings("PMD.ConstructorOnlyInitializesOrCallOtherConstructors")
+    public RustProject(final Path target) throws IOException {
+        this.footprint = new FtDefault(target);
+        this.dest = target;
+        this.dependencies = new HashSet<>();
+        this.footprint.save(
```

Comment:
@levBagryansky @Graur it's a bad idea in general to make evaluations inside constructors: https://www.yegor256.com/2015/05/07/ctors-must-be-code-free.html

Reply:
@yegor256 
Class `RustProject` must preserve its invariant: cargo project must be compilible after any method. A function initializing the object must set this invariant. That is why I guess it is bad idea to create a public method that set this invariant apart from constructor. It  would always follow constructor.
Thus, the constructor must be `private` since it does not set invariant.